### ISSUE TITLE
Hello World part 1 of n: basic functions, and pointers (do this after #8)

### ISSUE THREAD

**Issue by abstractmachines** [devinenoise/hello-world#5]
# Assignment 1: Create a couple of functions. One will be a "hello world" function that prints a string. Another function will be a function that returns a pointer to an integer of arrays. Call both of these functions from main().

## C++ assignment 1
- To get your editor set up, watch tutorials "VSCode C++ tutorial, Compile And Run On Windows", such as https://www.youtube.com/watch?v=DIw02CaEusY

- You can also work with this interactive shell to "get going": http://cpp.sh/

- C++ code samples: copypasta can be a good thing!
https://github.com/abstractmachines/cpp-algorithms
https://github.com/abstractmachines/OOP_Cpp_ComplexNumbers

> THE CODE:
One function should be called helloWorld() and should print out “Hello World.” The second member function should create a static array of integers and **return** that static array of integers (it should return a **pointer, to an array of ints**).

- TIP: First write a C++ function.
- Then print basic stuff out. LIke the tutorials you'll see on Youtube. PLAY around! This is a good start too! http://cpp.sh/
- After you can print basic stuff and write a function, stopping at each little step to compile and run your code one line at a time,
- But first, get super basic stuff working. Make one line work.

```
g++ myfile.cpp
```
^ I compiled this file!

```
// to run the file, Windows will show you how to run the actual executable file. It's hard to do in Windows.
```

## Definition of done

> the code
What it'll look like when things are complete (this is pseudocode):

```
import bunchastuff

void helloWorld() {
  print hello world string
}

int * arrayPrint() {
  create an array of 5 integers

  return that array (a pointer to that array).
}

int main () {

 helloWorld() // prints string

 // now create a pointer to an integer of arrays, assign it ( `=` ) to your function arrayPrint(), and print the array from here.


 return 0; // main returns an int.
}
```

> the documentation
- [x] Readme file should contain documentation for everything you're learning so far. Including pointers!

**Comment by abstractmachines** [comment]
# Function signature

This function returns an integer.

```
int thisReturnsInt();
```

**Comment by abstractmachines** [comment]
# Function body

> make sure you have a return statement to tell the function what to return.

```
int thisReturnsInt() {
    return 42;
}
```

**Comment by abstractmachines** [comment]
```
#include <iostream>

int numGroup() {
   return 42;
}

int main()
{
    std::cout << "Hello World!\n";

    // pause the string so the terminal window remains open
    // system("pause>0"); // commenting this out so it doesn't distract us.

   // one thing at a time.

   // let's call that function numGroup() from main. First, we have to have a data type inside of main() that's the same! int.
   int anIntInsideMain = numGroup(); // uses the return value of numGroup() function.

   std::cout << anIntInsideMain; // should be 42! 
    return 0;
}
```

once you've got that done, figure out how to return an array. That will be hard, because you'll have to learn pointers!

https://www.tutorialspoint.com/cplusplus/cpp_return_arrays_from_functions.htm

Return an array from numGroup.

**Comment by abstractmachines** [comment]
@devinenoise now that you've written a function that **returns** _void_ and prints out an array of integers, here:

https://github.com/devinenoise/hello-world/blob/main/hello-world.cpp#L36

Instead of that, I want you to return a function that **returns** _a pointer to an array of integers._ Make a new branch and PR.

For your next branch and PR, the return type for this function should be `int *`, a pointer to an array of integers.

Find out how to call that function from within your main function.

I left notes in issue #5 in comments - read the above about pointers.

@devinenoise

**Comment by abstractmachines** [comment]
Resolved mostly, by #10
